daemon/stats/collector_unix.go

```diff
@@ -54,15 +64,21 @@ func (s *Collector) getSystemCPUUsage() (uint64, error) {
 			for _, i := range parts[1:8] {
 				v, err := strconv.ParseUint(i, 10, 64)
 				if err != nil {
-					return 0, fmt.Errorf("Unable to convert value %s to int: %s", i, err)
+					return 0, fmt.Errorf("unable to convert value %s to int: %s", i, err)
 				}
 				totalClockTicks += v
+
 			}
-			return (totalClockTicks * nanoSecondsPerSecond) /
-				clockTicksPerSecond, nil
+
+			if totalClockTicks > maxClockTicks {
+				return 0, fmt.Errorf("number of clock ticks obtained from '%s' will overflow cpu stats calculations", procStatFile.Name())
+			}
+
+			return totalClockTicks * nanoSecondsPerTick, nil
 		}
 	}
-	return 0, fmt.Errorf("invalid stat format. Error trying to parse the '/proc/stat' file")
+
+	return 0, fmt.Errorf("invalid /proc/stat format. Error trying to parse the '%s' file", procStatFile.Name())
```

Comment:
Looking if it' s possible to make it a single sentence; perhaps something like;

```suggestion
	return 0, fmt.Errorf("invalid stat format while trying to parse '%s'", procStatFile.Name())
```